telluric/georaster.py

```diff
@@ -130,113 +208,114 @@ def _prepare_other_raster(one, other):
+    if one.band_names != other.band_names:
+        raise ValueError("rasters have no bands in common, use another merge strategy")
+
+    new_image = one.image.copy()
+    other_image = other.image
+
+    # The values that I want to mask are the ones that:
+    # * Were already masked in the other array, _or_
+    # * Were already unmasked in the one array, so I don't overwrite them
+    other_values_mask = (other_image.mask[0] | (~one.image.mask[0]))
+
+    # Reshape the mask to fit the future array
+    other_values_mask = other_values_mask[None, ...]
+
+    # Overwrite the values that I don't want to mask
+    new_image[~other_values_mask] = other_image[~other_values_mask]
+
+    # In other words, the values that I wanted to write are the ones that:
+    # * Were already masked in the one array, _and_
+    # * Were not masked in the other array
+    # The reason for using the inverted form is to retain the semantics
+    # of "masked=True" that apply for masked arrays. The same logic
+    # could be written, using the De Morgan's laws, as
+    # other_values_mask = (one.image.mask[0] & (~other_image.mask[0])
+    # other_values_mask = other_values_mask[None, ...]
+    # new_image[other_values_mask] = other_image[other_values_mask]
+    # but here the word "mask" does not mean the same as in masked arrays.
+
+    return _Raster(image=new_image, band_names=one.band_names)
+
+
+def _stack_bands(one, other):
+    # type: (_Raster, _Raster) -> _Raster
+    """Merges two rasters with non overlapping bands by stacking the bands.
 
-        return _merge(one, other, merge_strategy=MergeStrategy.INTERSECTION)
+    """
+    assert set(one.band_names).intersection(set(other.band_names)) == set()
+
+    # We raise an error in the bands are the same. See above.
+    if one.band_names == other.band_names:
+        raise ValueError("rasters have the same bands, use another merge strategy")
+
+    # Apply "or" to the mask in the same way rasterio does, see
+    # https://mapbox.github.io/rasterio/topics/masks.html#dataset-masks
+    # In other words, mask the values that are already masked in either
+    # of the two rasters, since one mask per band is not supported
+    new_mask = one.image.mask[0] | other.image.mask[0]
+
+    # Concatenate the data along the band axis and apply the mask
+    new_image = np.ma.masked_array(
+        np.concatenate([
+            one.image.data,
+            other.image.data
+        ]),
+        mask=[new_mask] * (one.image.shape[0] + other.image.shape[0])
+    )
+    new_bands = one.band_names + other.band_names
+
+    # We don't copy image and mask here, due to performance issues,
+    # this output should not use without eventually being copied
+    # In this context we are copying the object in the end of merge_all merge_first and merge
+    return _Raster(image=new_image, band_names=new_bands)
+
+
+def _merge_two(one, other, merge_strategy=MergeStrategy.UNION, silent=False):
```

Comment:
what about the silent, are there other ways to do it?

Reply:
I'm merging two functions into one here, because they had very similar behavior: `merge` and `merge_to_first`. We can keep them as separate functions, or just remove them and make `_merge_two` public. What do you think?

Reply:
you are right, although in the past we agreed to not to use this approach, in this case it is a reasonable solution

Reply:
OK, I will leave one function then